Automattic/wp-calypso
# Earn: Convert products component to TypeScript

## Proposed Changes

Converts the Earn > MembershipsProductsSection from class to function component, and from JS to TypeScript. 

## Testing Instructions



There should be no change in functionality before/after this PR. 

1) You will need a stripe connected site to test. 
2) Go to http://calypso.localhost:3000/earn/payments-plans/YOURDOMAIN. Create a product if you don't have one.
3) On a product, click the three dots on the right and select delete. Confirm the delete modal opens as before. 
4) Click 'Delete' on the modal, and confirm the plan is deleted.

## Pre-merge Checklist
